### PR TITLE
updated according to comments from 09.10.2024 from mentor

### PR DESCRIPTION
**Pull request by YauheniyaH**

### CODE REVIEW

File: src/main/java/pageobject_model/page/CloudGoogleComputeEnginePage.java [hunk 1]
```diff
@@ -7,51 +7,70 @@
...
-import java.util.logging.FileHandler;
-import java.util.logging.Level;
 import java.util.logging.Logger;
 
 import org.apache.commons.io.FileUtils;
 
+import static java.lang.String.format;
+import static java.util.logging.Level.INFO;
+import static org.openqa.selenium.OutputType.FILE;
+
```
Comment (ADrema) [inline src/main/java/pageobject_model/page/CloudGoogleComputeEnginePage.java:16]: one empty line is enough

File: src/main/java/pageobject_model/test/WebDriverCloudGoogleTest.java [hunk 2]
```diff
@@ -35,14 +35,19 @@ public void browserSetup() throws IOException {
     public void calculateMonthlyRentForParameters() throws InterruptedException {
         computeEnginePage.inputNumberOfInstances("4");
         computeEnginePage.optionalPopUpClose();
+
         computeEnginePage.selectDropdownValue("OS","'Free: Debian, CentOS, CoreOS, Ubuntu or BYOL (Bring Your Own License)'");
```
Comment (ADrema) [inline src/main/java/pageobject_model/test/WebDriverCloudGoogleTest.java]: OS, machineType, machineType can be moved into separate class as Dropdown values

File: src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java [hunk 3]
```diff
@@ -14,27 +16,41 @@
...
         driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
         driver.manage().window().maximize();
-        CloudGoogleComputeEnginePage computeEnginePage = new CloudGoogleComputeEnginePage(driver);
+        computeEnginePage = new CloudGoogleComputeEnginePage(driver);
         computeEnginePage.openPage();
+    }
 
+    @Test
+    public void seleniumGridTest() {
+        assertEquals("Compute Engine", computeEnginePage.pageTitle.getText());
```
Comment (ADrema) [inline src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:45]: please open the page inside the test

File: src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java [hunk 4]
```diff
@@ -14,27 +16,41 @@
...
+    private static final String GRID_HOST = "http://localhost:4444/wd/hub";
 
-    @Test
-    public void seleniumGridTest() throws URISyntaxException, IOException {
+    private WebDriver driver;
+
+    private CloudGoogleComputeEnginePage computeEnginePage;
+
+    @BeforeMethod(alwaysRun = true)
+    public void browserSetup() throws IOException, URISyntaxException {
```
Comment (ADrema) [inline src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:30]: please move current tests to selenium Grid

File: src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java [hunk 5]
```diff
@@ -14,27 +16,41 @@
...
 
+    @Test
+    public void seleniumGridTest() {
+        assertEquals("Compute Engine", computeEnginePage.pageTitle.getText());
+    }
 
+    @AfterMethod(alwaysRun = true)
+    public void browserTearDown() {
         driver.quit();
+        driver = null;
```
Comment (ADrema) [inline src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:51]: Do you really need it?

Comment (YauheniyaH) [reply src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:51]: This helps ensure that the browser session is completely closed and the memory is released.

File: src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java (hunk 3 above)
Comment (YauheniyaH) [reply src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:45]:
```suggestion
        computeEnginePage.openPage();
        assertEquals("Compute Engine", computeEnginePage.pageTitle.getText());
```

File: src/main/java/pageobject_model/page/CloudGoogleComputeEnginePage.java (hunk 1 above)
Comment (YauheniyaH) [reply src/main/java/pageobject_model/page/CloudGoogleComputeEnginePage.java:16]: done

File: src/main/java/pageobject_model/test/WebDriverCloudGoogleTest.java (hunk 2 above)
Comment (YauheniyaH) [reply src/main/java/pageobject_model/test/WebDriverCloudGoogleTest.java]: will move to enum inside DropDownObject class

File: src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java (hunk 3 above)
Comment (YauheniyaH) [reply src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:45]: will be added in next commit like suggested

File: src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java (hunk 4 above)
Comment (YauheniyaH) [reply src/main/java/pageobject_model/test/WebDriverSeleniumGridTest.java:30]: added to @BeforeMethod driver creation steps using RemoteWebDriver and DesiredCapabilities

**Pull request merged**